src/main/java/com/cloudbees/jenkins/support/impl/JenkinsLogs.java

```diff
@@ -112,9 +109,18 @@ protected void printTo(PrintWriter out) throws IOException {
                                     out.println("N/A");
                                 } else {
                                     try {
-                                        List<LogRecord> records = needHack
-                                                ? SlaveLogFetcher.getLogRecords(computer)
-                                                : computer.getLogRecords();
+                                        List<LogRecord> records = null;
+                                        if (needHack) {
+                                            if (computer.getChannel() != null) {
```

Comment:
:bug: race condition here... the channel could go null before the next line, you need to capture it in a local variable before the null check